strict-containers/src/Data/FingerTree/Strict.hs

```diff
@@ -73,9 +73,8 @@ infixl 5 |>
 -- wrap the original "Data.FingerTree" functions while forcing the provided
 -- value to WHNF.
 newtype StrictFingerTree v a = SFT {fromStrict :: FT.FingerTree v a}
-  deriving (Eq, Ord, Show)
-
-deriving newtype instance Foldable (StrictFingerTree v)
+  deriving (Eq, Ord, Show, Generic)
+  deriving newtype (Foldable, Semigroup, Monoid)
 
 deriving newtype instance (Measured v a) => Measured v (StrictFingerTree v a)
```

Comment:
What about moving `Measured v` into the `deriving newtype (...)` just above like `Foldable`?